packages/sanity/src/core/studio/router/helpers.ts

```diff
@@ -125,6 +125,7 @@ export function resolveIntentState(
 
     const nextUrlState: Record<string, unknown> = {
       ...prevState,
+      _searchParams: nextState._searchParams,
```

Comment:
Wondering, what's the reason to use `_searchParams` instead of `searchParams`

Reply:
This is because the router expects the search params to be in a `_searchParams` property.
I've added an update to the `RouterEvent` interface to reflect that the state returned from this function should be a RouterState.
```diff
export interface RouterStateEvent {
  type: 'state'
+ state: RouterState
- state: Record<string, unknown> 
  isNotFound: boolean
}
```

RouterState is the following:
`export type RouterState = Record<string, unknown> & {_searchParams?: SearchParam[]}
`

Reply:
https://github.com/sanity-io/sanity/pull/7095/commits/646f433231a5f0ca8f0fd9baacc195e539c4257e

Reply:
understood, thanks for confirming